log4j-api/src/main/java/org/apache/logging/log4j/internal/LogBuilder.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+package org.apache.logging.log4j.internal;
+
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.message.Message;
+
+/**
+ * Class Description goes here.
+ */
+public interface LogBuilder {
+
+    public static final LogBuilder INSTANCE = new LogBuilder() {};
+
+    default LogBuilder withMarker(Marker marker) {
+        return this;
+    }
+
+    default LogBuilder withThrowable(Throwable throwable) {
+        return this;
+    }
+
+    default LogBuilder withLocation() {
+        return this;
+    }
+
+    default LogBuilder withLocation(StackTraceElement location) {
+        return this;
+    }
+
+    default LogBuilder withMessage(Message msg) {
+        return this;
+    }
+
+    default LogBuilder withMessage(String msg) {
+        return this;
+    }
+
+    default LogBuilder withMessage(Object msg) {
+        return this;
+    }
+
+
+    default LogBuilder withParameters(Object... params) {
```

Comment:
How does this interact with `withMessage(Message)`? Do we expect to overload this method with variable numbers of arguments similarly to existing logger.<level> methods?

Would it be possible to use a single method which adds a parameter?

```java
log.info()
  .message("{}, {}!)
  .parameter("Hello")
  .parameter("World")
  .log();
```

Reply:
The parameters are passed to the message factory along with the format string when constructing a message. If a Message is supplied the parameters would be ignored.

You can specify multiple parameters on the parameters method and you can invoke the parameters method as many times as you want. I need to add another parameters method that accepts a supplier.